src/hooks/useLinearPrimaryTicks.ts

```diff
@@ -0,0 +1,98 @@
+import type { ScaleContinuousNumeric } from 'd3-scale';
+import { MutableRefObject, useEffect, useMemo, useState } from 'react';
+
+import { textDimentions } from '../utils';
+
+type Directions = 'horizontal' | 'vertical';
+
+export interface PrimaryLinearTicks {
+  label: string;
+  position: number;
+}
+
+interface Options {
+  tickFormat?: (d: number) => string;
+  scientificNotation?: boolean;
+  minSpace?: number;
+}
+
+const TEST_HEIGHT = '+1234567890';
+const MAX_ITERATION = 20;
+
+export function useLinearPrimaryTicks<
+  Scale extends ScaleContinuousNumeric<number, number>
+>(
+  scale: Scale,
+  direction: Directions,
+  ref: MutableRefObject<SVGGElement | null>,
+  options: Options = {},
+): PrimaryLinearTicks[] {
+  const [ticks, setTicks] = useState<number[]>([]);
+
+  const range = scale.range();
+  if (!range) throw new Error('Range needs to be specified');
+
+  const domain = scale.domain();
+  if (!domain) throw new Error('Domain needs to be specified');
+
+  const { scientificNotation = false, minSpace = 4 } = options;
+  const tickFormat: (d: number) => string = useMemo(
+    () =>
+      options.tickFormat || scientificNotation
+        ? (x) => x.toExponential(2)
+        : (x) => JSON.stringify(x),
```

Comment:
Why `JSON.stringify`?

Reply:
Isn't it the standard? which one will you use?

Reply:
If what you want to do is convert the number to string, `String(x)`